### PR TITLE
Lionelb/refactor api annuaire

### PR DESCRIPTION
**Pull request by lionelB**
refactor api to separate files 
I update `packages.json` to avoid jest error/warning on `*.mock.js` since they are not test files.
I mock `node-fetch` module to return the location of loriol (near an valence  annuaire document)
I've also updated the elastic search data to have two annuaire document that are farther to only retrieve one document depending on location

### CODE REVIEW

**Review by revolunet — APPROVED** [review]

File: packages/code-du-travail-api/src/server/routes/__tests__/annaire.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+test("should return all the records around loriol", async () => {
+  fetch.mockResolvedValue({ json: () => addresses.payload });
+
+  const response = await request(app.callback()).get(
+    `/api/v1/annuaire/search?q=${addresses.query}`
+  );
+  expect(response.status).toBe(200);
+  expect(response.body.hits).toMatchSnapshot();
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/__tests__/annaire.spec.js:30]: Missing `expect` on fetch.
We should ensure that it's called. 
We don't need to mock it if not

File: packages/code-du-travail-api/src/server/routes/__tests__/api-adresse.mock.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module.exports = {
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/__tests__/api-adresse.mock.js]: It looks like a `json` file to me no ?

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const elasticsearchClient = require("../../conf/elasticsearch.js");
+const getAnnuaireBody = require("./annuaire.elastic");
+
+const router = new Router({ prefix: API_BASE_URL });
+const index = process.env.ELASTICSEARCH_ANNUAIRE_INDEX || "cdtn_annuaire";
+const routeName = "/annuaire";
+
+// todo @lionelb move to constant file
+const defaultMaxSearchDistance = "30km";
+const adresseApiEndPoint = "https://api-adresse.data.gouv.fr/search";
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/annuaire/index.js:14]: IMO those const looks like configurations.
We might consider extracting them from the code.

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * @example
+ * http://localhost:1337/api/v1/annuaire/search?coord=lon:lat
+ * http://localhost:1337/api/v1/annuaire/search?q=address to search
+ *
+ * @param {string} :base coord or address for geo search
+ * @returns {Object} Result.
+ */
+router.get(`${routeName}/search`, async ctx => {
+  const { coord, q } = ctx.request.query;
+  const results = searchFromCoord(coord) || (await searchFromQuery(q));
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/annuaire/index.js]: :heart: that 
Are the `()` parenthesizes really required ?

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    ctx.body = await elasticsearchClient.search({
+      index,
+      body
+    });
+  } else {
+    ctx.body = { total: 0, hits: { hits: [] } };
+  }
+});
+
+function searchFromCoord(coord) {
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/annuaire/index.js:46]: If we want to go full functional `searchFromQuery` should be curried to take the `adresseApiEndPoint` as first arguments

```js
async function searchFromQuery(adresseApiEndPoint){ 
  return query => {
    if (!query) {
      return false;
    }
  // [...]
```

 It's looking very good already :)

**Review by douglasduteil — CHANGES_REQUESTED** [review]
Love this coding style :100:

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js (hunk 3 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/annuaire/index.js:14]: yup, move it to a separate file ie `route.config.json` // `api.config.{json|js}`

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js (hunk 4 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/annuaire/index.js]: > Are the `()` parenthesizes really required ?
prettier :)

File: packages/code-du-travail-api/src/server/routes/annuaire/index.js (hunk 5 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/annuaire/index.js:46]: currying is not needed for now, so I only pass endpoint as a parameter so we could curry it later if we need it

**Comment by lionelB** [comment]
@douglasduteil updated !

File: packages/code-du-travail-api/package.json [hunk 6]
```diff
@@ -52,8 +52,13 @@
     "supertest": "^3.3.0"
   },
   "jest": {
+    "testPathIgnorePatterns": [
+      "node_modules",
+      ".mock.js$"
+    ],
     "collectCoverageFrom": [
       "src/**/*.js",
+      "!src/**/*.mock.js",
```
Comment (douglasduteil) [inline packages/code-du-travail-api/package.json]: Still required now that mocks are json files ?

Comment (lionelB) [reply packages/code-du-travail-api/package.json]: no no no I update

**Comment by lionelB** [comment]
@douglasduteil 🆙

**Review by douglasduteil — APPROVED** [review]
Super

**Pull request merged**